src/utils/update/beneficial_owners_managing_officers_data_fetch.ts

```diff
@@ -37,38 +36,30 @@ const initialiseBoAndMoUpdateAppData = (appData: ApplicationData) => {
 };
 
 export const retrieveBeneficialOwners = async (req: Request, appData: ApplicationData) => {
-  const transactionId = appData.transaction_id;
-  const overseasEntityId = appData.overseas_entity_id;
   const pscs: CompanyPersonsWithSignificantControl = await getCompanyPsc(req, appData[EntityNumberKey] as string);
 
-  let boPrivateData: BeneficialOwnersPrivateData = {
-    boPrivateData: []
-  };
-
-  if (overseasEntityId && transactionId){
-    try {
-      boPrivateData = await getBeneficialOwnerPrivateData(req, transactionId, overseasEntityId) as BeneficialOwnersPrivateData;
-    } catch (error) {
-      logger.errorRequest(req, "Private Beneficial Owner details could not be retrieved for overseas entity " + appData.entity_number);
-    }
-  }
+  const boPrivateData = await getBoPrivateData(req, appData);
 
   if (pscs) {
     for (const psc of (pscs.items || [])) {
       logger.info("Loaded psc " + psc.kind);
-      if (psc.ceasedOn === undefined && psc.kind === "individual-beneficial-owner") {
-        const individualBeneficialOwner = mapPscToBeneficialOwnerTypeIndividual(psc);
-        mapBeneficialOwnerDetails(individualBeneficialOwner, boPrivateData!);
-        logger.info("Loaded individual Beneficial Owner " + individualBeneficialOwner.id + " is " + individualBeneficialOwner.first_name + ", " + individualBeneficialOwner.last_name);
-        appData.update?.review_beneficial_owners_individual?.push(individualBeneficialOwner);
-      } else if (psc.ceasedOn === undefined && psc.kind === "corporate-entity-beneficial-owner"){
-        const beneficialOwnerOther = mapPscToBeneficialOwnerOther(psc);
-        logger.info("Loaded Beneficial Owner Other " + beneficialOwnerOther.id + " is " + beneficialOwnerOther.name);
-        appData.update?.review_beneficial_owners_corporate?.push(beneficialOwnerOther);
-      } else if (psc.ceasedOn === undefined && psc.kind === "legal-person-beneficial-owner") {
-        const beneficialOwnerGov = mapPscToBeneficialOwnerGov(psc);
-        logger.info("Loaded Beneficial Owner Gov " + beneficialOwnerGov.id + " is " + beneficialOwnerGov.name);
-        appData.update?.review_beneficial_owners_government_or_public_authority?.push(beneficialOwnerGov);
+      if (psc.ceasedOn === undefined){
+        if (psc.kind === "individual-beneficial-owner") {
+          const individualBeneficialOwner = mapPscToBeneficialOwnerTypeIndividual(psc);
+          individualBeneficialOwner.usual_residential_address = mapBoPrivateAddress(boPrivateData, individualBeneficialOwner.ch_reference!);
+          logger.info("Loaded individual Beneficial Owner " + individualBeneficialOwner.id + " is " + individualBeneficialOwner.first_name + ", " + individualBeneficialOwner.last_name);
+          appData.update?.review_beneficial_owners_individual?.push(individualBeneficialOwner);
+
```

Comment:
Just a thought, we could extract this out to something like `mapIndividualBeneficialOwner(psc: CompanyPersonWithSignificantControl, boPrivateData: BeneficialOwnersPrivateData | undefined, appData: ApplicationData)`. Maybe there's a more suitable name.

Reply:
Trying this now, but tests are failing at the moment, should I keep looking for a solution, or is the current fine?

Reply:
a bit odd, they seem to have passed now , will push

Reply:
seems to not be working with if (psc.ceasedOn === undefined) {
continue; }